ddtrace/contrib/starlette/patch.py

```diff
@@ -55,13 +60,17 @@ def patch():
     starlette._datadog_patch = True
 
     _w("starlette.applications", "Starlette.__init__", traced_init)
+    Pin().onto(starlette)
```

Comment:
Why is this necessary?

Reply:
@mabdinur can correct me if I'm wrong, but my understanding was that we need this for the `_trace_background_tasks()` function when we're trying to create the span links between the original task and the background task. We pass in the entire `starlette`/`fastapi` module, so we needed the pin to be associated with them at the module level.

Reply:
Yup. It's a bug. The starlete integration is not creating a pin but it is referencing it in different places. 

We could've pull this one line out into its own PR but I got lazy